### PR TITLE
fix: handle message hashing correctly

### PR DESCRIPTION
**Pull request by kaladinlight**
this update handles the following message formats correctly now:
- utf8 string
- hex encoded utf8 string
- keccak256 encoded string
- raw bytes

### CODE REVIEW

File: packages/hdwallet-native/src/crypto/isolation/adapters/ethereum.ts [hunk 1]
```diff
@@ -69,10 +69,15 @@ export class SignerAdapter {
   }
 
   async signMessage(messageData: ethers.Bytes | string, addressNList: core.BIP32Path): Promise<string> {
-    const messageDataBuf =
-      typeof messageData === "string"
-        ? Buffer.from(messageData.normalize("NFKD"), "utf8")
-        : Buffer.from(ethers.utils.arrayify(messageData));
+    const messageDataBuf = (() => {
+      if (typeof messageData === "string") {
+        if (ethers.utils.isHexString(messageData)) {
```
Comment (gomesalexandre) [inline packages/hdwallet-native/src/crypto/isolation/adapters/ethereum.ts:74]: 🧠

**Review by gomesalexandre — APPROVED** [review]
Untested but conceptually LGTM

**Review by BitHighlander — APPROVED** [review]
:shipit:

**Review by lmyslinski — APPROVED** [review]

**Review by gomesalexandre — APPROVED** [review]
LGTM diff-wise, stamping as this has been tested by ops in prod with an alpha hdwallet release

**Pull request merged**